util/flocks/src/tfrwlock/mod.rs

```diff
@@ -0,0 +1,56 @@
+pub mod write_guard;
+pub mod read_guard;
+
+pub use write_guard::TfrwLockWriteGuard;
+pub use read_guard::TfrwLockReadGuard;
+
+use tokio::sync::RwLock;
+use crate::frwlock::{FrwLock, FrwLockError};
+use rustix::fd::AsFd;
+use thiserror::Error;
+
+
+#[derive(Debug, Error)]
+pub enum TfrwLockError {
+    #[error("Lock is not available")]
+    LockNotAvailable,
+    #[error("File error: {0}")]
+    FileError(#[from] std::io::Error),
+}
+
+impl From<FrwLockError> for TfrwLockError {
+    fn from(e: FrwLockError) -> Self {
+        match e {
+            FrwLockError::LockNotAvailable => TfrwLockError::LockNotAvailable,
+            FrwLockError::FileError(e) => TfrwLockError::FileError(e),
+        }
+    }
+}
+
+/// A file-based read-write lock.
+/// This only mutually excludes processes trying to violate the lock, not the same process--which is not considered contention.
+/// If you want to prevent contention within the same process, you should wrap this in your preferred synchronization primitive.
+pub struct TfrwLock<T: AsFd> {
```

Comment:
What todes the T stand for here? In the name `Tfrw`

Reply:
Thread